pkg/deploy/controller/deployment/factory.go

```diff
@@ -122,6 +126,15 @@ func (factory *DeploymentControllerFactory) Create() controller.RunnableControll
 	}
 }
 
+// setPodAttributes updates the deployer pod attributes based on the deployment strategy configuration
+func (factory *DeploymentControllerFactory) setPodAttributes(strategy *deployapi.DeploymentStrategy, pod *kapi.Pod) {
+	switch strategy.Type {
+	case deployapi.DeploymentStrategyTypeCustom:
+		util.MergeInto(pod.Labels, strategy.Labels, 0)
```

Comment:
I'd like at least one other person (@smarterclayton @liggitt @kargakis) to validate my premise (that we can apply labels/annotations to all deployer pods). If all agree, you can remove the `setPodAttributes` abstraction.